connectors/register.sh

```diff
@@ -93,14 +93,39 @@ open_dataset_api_ports(){
 }
 
 register_connectors() {
-    # list all normal files in distributions directory
-    for connector in $(ls -1 distributions/*.tar.gz); do
-        echo "\nRegistering connector: $connector"
-        curl --progress-bar --location "localhost:$dataset_api_port/v2/connector/register" \
-        --header 'Content-Type: multipart/form-data' \
-        --form "file=@$connector" | cat
-        rm -rf $connector
-    done
+  echo "📦 Starting GitHub-based connector registration..."
+  mkdir -p distributions
+
+  # List of connectors as "repo-name:asset-filename"
+  connectors=(
+    "kafka-connector:kafka-connector.tar.gz"
+    "jdbc-connector:jdbc-connector.tar.gz"
+    "object-store-connector:object-store-connector.tar.gz"
+    "debezium-connector:debezium-connector.tar.gz"
+  )
+
+  for entry in "${connectors[@]}"; do
+    repo_name="${entry%%:*}"
+    asset_name="${entry##*:}"
+    download_url="https://github.com/Sanketika-Obsrv/${repo_name}/releases/latest/download/${asset_name}"
+    output_path="distributions/${asset_name}"
+
+    echo "⬇️  Downloading $asset_name from $repo_name..."
+    if curl -L --fail "$download_url" -o "$output_path"; then
+      echo "Downloaded: $output_path"
+    else
+      echo "Failed to download: $asset_name from $repo_name"
+      continue
+    fi
+
+    echo "Registering: $asset_name to Dataset API..."
+    curl --progress-bar --location "http://localhost:$dataset_api_port/v2/connector/register" \
```

Comment:
Why localhost? How is it going to access the API if I execute the installation script from the VM.